Tests/LoggingTests/LoggingTest.swift

```diff
@@ -357,6 +357,25 @@ class LoggingTest: XCTestCase {
                                                    "nested-list": ["l1str", ["l2str1", "l2str2"]]])
     }
 
+    func testWithMetadata() {
```

Comment:
Could we add three more test cases:
1. Where the logger has some existing metadata and we add new metadata without any overlapping keys
2. Same as above but with overlapping keys
3. Adding more than once with overlapping keys

Reply:
Added to existing `testMergeMetadata` function.